### PR TITLE
Add short-iso-precise for journalctl output

### PR DESCRIPTION
**Pull request by ianw**
This adds a short-iso-precise option for journalctl output.  It is as for
short-iso, but includes microseconds.

To motivate this change -- in upstream OpenStack CI devstack we were running all the services during testing under "screen".  One screen per service, which corresponds to one log file per dameon.  Each service would just be doing it's own logging to stdout (captured to file), which included a high-precision timestamp in ISO format.

We have recently switched to running services under systemd and storing their logs in the journal.  Of course everything is in the journal ... however, we still write out individual log files for each daemon as part of the CI runs and for log storage.  this is important because there's tooling around these existing logs and, well, people are used to them.  

So it seems we really only have two options when writing out these log files from the journal

1. we can use "--short-precise" to get microseconds ... but then we get the syslog format "Mon XX ...".  This is really annoying and the string month names make parsing unnecessarily hard.
2. use "--short-iso" to get more standard timestamps, but then drop sub-second output

Both of these are a regression from where we were when the daemons just logged their own output with their own timestamps.

I feel like short-iso probably should just have microseconds, but I imagine that ship has sailed backwards-compat wise.  Thus I think a --short-iso-precise will be helpful in general.  A small example is below.

Thanks

 ```
[iwienand@fedora19 tmp]$ ~/programs/systemd/journalctl -D . -o short-iso-precise | head 
-- Logs begin at Thu 2017-05-04 03:19:15 AEST, end at Thu 2017-05-04 03:44:21 AEST. --
2017-05-04T03:19:15.060209+1000 ubuntu systemd-journald[271]: Runtime journal (/run/log/journal/) is 8.0M, max 79.8M, 71.8M free.
2017-05-04T03:19:15.071261+1000 ubuntu systemd-journald[271]: System journal (/var/log/journal/) is 8.0M, max 1.3G, 1.3G free.
2017-05-04T03:19:15.072065+1000 ubuntu systemd-journald[271]: Time spent on flushing to /var is 705us for 2 entries.
2017-05-04T03:19:15.072087+1000 ubuntu kernel: Initializing cgroup subsys cpuset
```

### CODE REVIEW

File: src/shared/logs-show.c [hunk 1]
```diff
@@ -280,6 +282,16 @@ static int output_timestamp_realtime(FILE *f, sd_journal *j, OutputMode mode, Ou
                         }
                         break;
 
+                case OUTPUT_SHORT_ISO_PRECISE:
+                        /* No usec in strftime, so we leave space and copy over */
+                        if (strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%S.xxxxxx%z", gettime_r(&t, &tm)) <= 0) {
+                                log_error("failed for format ISO-precise time");
```
Comment (keszybz) [inline src/shared/logs-show.c]: Error messages start with a capital letter. Also, "failed for" should be "failed to". Please fix the new message and also the one above.

File: src/shared/logs-show.c [hunk 2]
```diff
@@ -280,6 +282,16 @@ static int output_timestamp_realtime(FILE *f, sd_journal *j, OutputMode mode, Ou
...
+                                return -EINVAL;
+                        }
+                        snprintf(usec, 7, "%06"PRI_USEC, x % USEC_PER_SEC);
```
Comment (keszybz) [inline src/shared/logs-show.c]: `xsprintf(usec, "%06"PRI_USEC, x % USEC_PER_SEC)`.

File: src/shared/logs-show.c (hunk 1 above)
Comment (ianw) [reply src/shared/logs-show.c]: Done

File: src/shared/logs-show.c (hunk 2 above)
Comment (ianw) [reply src/shared/logs-show.c]: Done

**Comment by keszybz** [comment]
Oops, I missed one more necessary change: please amend --help output in the other places:

```
src/journal/journalctl.c:               "  -o --output=STRING       Change journal output mode (short, short-precise,\n"
src/login/loginctl.c:               "  -o --output=STRING       Change journal output mode (short, short-monotonic,\n"
src/machine/machinectl.c:               "  -o --output=STRING          Change journal output mode (short,\n"
src/systemctl/systemctl.c:               "  -o --output=STRING  Change journal output mode (short, short-precise,\n"
```

**Comment by ianw** [comment]
@keszybz : thank you for noticing that; it seems those have got a out of sync and missed short-iso, so missed my grepping!  All synced in next change.

**Pull request merged**